content/posts/hugo-blog-google-cloud-storage.md

```diff
@@ -0,0 +1,112 @@
+
+These steps are just the basics. For more options and details, check [the documentation](https://cloud.google.com/storage/docs/hosting-static-website).
+
+
+
+* Configure gcloud to use your cloud project:
+```
+ gcloud config set project my-project
+```
+replace `my-project` with your own
+* Enable the services you'll use for this solution
+```
+gcloud services enable compute.googleapis.com storage-component.googleapis.com cloudbuild.googleapis.com 
+```
+
+* Create a storage bucket to hold the static website files:
+```
+gcloud storage buckets create gs://my-blog-files --location=us-central1
+```
+replace `my-blog-files` with a name for your storage bucket
+
+replace `us-central1` with a [different zone](https://cloud.google.com/compute/docs/regions-zones#available) if you prefer
+* Make the storage bucket public for reading: 
+```
+gcloud storage buckets add-iam-policy-binding  gs://my-blog-files --member=allUsers --role=roles/storage.objectViewer
+```
+replace `my-blog-files` with the name for your storage bucket
+
+## Set up your load balancer and SSL certificate
+Next, set up your load balancer and SSL certificate for your domain.  I've found this easiest to do [in the console](https://cloud.google.com/storage/docs/hosting-static-website#lb-ssl).
+
+{{< figure src="/images/gcp-blog/new-cert.png" width="400px" title="Creating a new certificate for the site" >}}
+{{< figure src="/images/gcp-blog/finalize-lb.png" title="Setting up a load balancer with SSL" >}}
+
+## Setting up your Hugo site
+
+Now the fun part.  I'll keep this hugo site _very_ simple.
+
+
+* Make a new hugo site:
+```
+hugo new site wasd
+```
+replace `wasd` with the name of your site, and `cd` into that directory when you've run the command.
+
+* Initialize a new hugo module. This makes it much easier to integrate themes and keep them up to date:
+```
+hugo mod init github.com/arbrown/wasd
+```
+* Add a theme to the modules section of `hugo.toml`:  
+```
+[module]
+  [[module.imports]]
+    path = "github.com/vaga/hugo-theme-m10c"
+```
+I'm using [m10c](https://github.com/vaga/hugo-theme-m10c) for this site, but feel free to check out [other options](https://themes.gohugo.io/)
+* Update your hugo module with the recently added theme:
+```
+hugo mod get -u
+```
+
+* Run hugo's built-in server to see your new website! 
+```
+hugo server
+```
+
+## Deploying to Google Cloud Storage
+Now leverage hugo's built-in deployment feature to push your site's files to your cloud storage bucket. `hugo deploy` uses the Google Cloud Storage API to check which files have changed (or been added or deleted) since the last time you uploaded them, and then does all the work to update your site with any new content you've added.
+
+Add the following section to `hugo.toml`:
+```
+[deployment]
+  [[deployment.targets]]
+    name = "storage_bucket"
+    URL = "gs://wasd-blog"
```

Comment:
Didn't know about this, awesome! Could you link to the hugo docs about this so I can read more?

Reply:
Done! I definitely _meant_ to do that originally.

https://gohugo.io/hosting-and-deployment/hugo-deploy/